runtime/lua/vim/filetype.lua

```diff
@@ -19,6 +19,10 @@ local function starsetf(ft)
   }}
 end
 
+local function getline(buf, line)
+  return api.nvim_buf_get_lines(buf, line-1, line, true)[1]
+end
```

Comment:
This needs to be a protected call, because the line number could be out of range.

```suggestion
local function getline(bufnr, lnum)
  local ok, lines = pcall(api.nvim_buf_get_lines, bufnr, lnum-1, lnum, true)
  if not ok then
    return ""
  end
  return lines[1]
end
```

Reply:
You mean in case the file is empty?

Wouldn't it be enough to set the last argument to `false`?

Reply:
Hmm, the `nvim_buf_get_lines` docs say that `false` means out of bounds indices are "clamped to the nearest value", which implies that e.g. if line 2 doesn't exist, it will return whatever's on line 1 (which we do not want). But in practice it looks like it just returns an empty list. So that should work.